### DISCUSSION TITLE
Nested YAML

### DISCUSSION THREAD

**Discussion by ararrocks** [SkepticMystic/breadcrumbs#577] [Q&A]
hi, is there a way to utilise nested YAML too?

i have extensive use of nested yaml like:
relations:
  - relation-type: Cousin
    relation-link: "[[Arne]]"
  - relation-type: Wife
    relation-link: "[[Daniela]]"
  - relation-type: Child
    relation-link: Valentin
  - relation-type: Child
    relation-link: Johanna
    
or
jobs:
  - job-name: Driver
    job-company: "[[Campany 1]]"
   active: true
  - job-name: Booker
    job-company: "[[Comapny 2]]"
    active: false
    
    
is there a way to translate this into breadcrumbs?

**Comment by ChristinWhite** [comment]
Did you ever figure out if this was possible? I'm thinking about trying to use the same type of structure.

**Reply by ArneNostitz** [reply]
No i Never did - but would be happy to :-) i am not using breadcrumbs atm. but the nested yaml i use extensively

**Reply by ChristinWhite** [reply]
Yeah, I am as well, I'd not tried to nest significant edge fields though before. It doesn't seem like it would be super difficult to implement, I may have to take a crack at it when I get a little free time.

**Reply by michaelpporter** [reply]
Nested yaml is not supported in frontmatter. It turns in to json. It might be possible, what would a use case be?

**Reply by ArneNostitz** [reply]
hi michael :) its so long ago - i cant really remember ... ;-) 

but i think that i wanted to use my existing nested yaml that defines relation.cousin as 
relation
   - type: cousin
     link: [[asdasd]]
     
     
or in a relation to jobs - that would automatically create links to a company hiararchy ... but i would have to think and look again :) cheers

**Comment by michaelpporter** [comment]
Short answer: Breadcrumbs reads links from **flat** frontmatter properties, and nested objects like `relations: [{relation-type, relation-link}]` aren't supported — and won't be.

The reason is Obsidian itself moved away from nested frontmatter. Since the Properties UI landed, only flat scalars and lists are first-class: nested objects can't be edited in the UI, and Obsidian's link resolution (`frontmatterLinks`) only indexes links in flat properties, not links buried inside nested arrays. A nested-YAML feature would produce data you can't manage in-app and links the vault can't resolve.

The good news is flat typed-link fields already express exactly what nested pairs would — one property per relation type:

```yaml
---
parent: "[[Some Note]]"
friend: "[[Another Note]]"
---
```

Each becomes a typed edge, fully editable in the Properties UI and resolved natively. So there's no capability lost by staying flat. Closing as resolved — thanks!
